test/test_torch.py

```diff
@@ -6970,18 +6970,6 @@ def _test_logical(self, device, op, a_, b_, expected_res_):
             for other_dtype in torch.testing.get_all_dtypes():
                 b = torch.tensor(b_, dtype=other_dtype, device=device)
 
-                # Skip bfloat16 on CUDA. Remove this after bfloat16 is supported on CUDA.
-                if device.startswith('cuda') and torch.bfloat16 in (dtype, other_dtype):
-                    with self.assertRaises(RuntimeError):
-                        getattr(a, op)(b)
-                    continue
-                # TODO Remove this skipping after bfloat16 can be handled nicely with other dtypes.
-                # Skip only if either dtype or other_dtype is bfloat16.
```

Comment:
This part of the test can be updated but you'll still need to exclude float16 x bfloat16 and verify you're not trying to compare in bfloat16 on CUDA. You can probably do this by calling promote types, catching an error if one occurs, and then testing that the result dtype is not bfloat16 if the device_type is CUDA.

If you do that, by the way, please check the device type correctly with `self.device_type == 'cuda'`, the above `device.startwith()` check is not a pattern we want to encourage.

Reply:
The device should not be cuda if the test proceeds to this line. I've addressed other comments on this line.